File: hamming-distance/__tests__/hammingDistance.test.js
const { hammingDistance } = require('../hammingDistance');

describe('Given two strings of equal length, return the number of different characters between them', () => {
    test('Given "string" & "string", return 0', () => {
        const string1 = 'string';
        const string2 = 'string';
        expect(hammingDistance(string1, string2)).toBe(0);
    });
    test('Given "I like turtles" & "I like turkeys", return 3', () => {
        const string1 = 'I like turtles';
        const string2 = 'I like turkeys';
        expect(hammingDistance(string1, string2)).toBe(3);
    });
    test('Given "a man a plan a canal" & "a man a plan sobanal", return 3', () => {
        const string1 = 'a man a plan a canal';
        const string2 = 'a man a plan sobanal';
        expect(hammingDistance(string1, string2)).toBe(3);
    });
    test('Given "hamming and cheese" & "Hamming and Cheese", return 2', () => {
        const string1 = 'hamming and cheese';
        const string2 = 'Hamming and Cheese';
        expect(hammingDistance(string1, string2)).toBe(2);
    });
    test('Given "espresso" & "Expresso", return 2', () => {
        const string1 = 'espresso';
        const string2 = 'Expresso';
        expect(hammingDistance(string1, string2)).toBe(2);
    });
    test('Given "old father, old artificer" & "of my soul the uncreated", return 24', () => {
        const string1 = 'old father, old artificer';
        const string2 = 'of my soul the uncreated';
        expect(hammingDistance(string1, string2)).toBe(24);
    });
});



//https://www.codewars.com/kata/5410c0e6a0e736cf5b000e69

File: python-venv/hamming_distance/__tests__/hamming_distance_test.py
from kata_function.hamming_distance import hamming_distance

def test_1():
    string1 = 'string'
    string2 = 'string'
    assert hamming_distance(string1, string2) == 0

def test_2():
    string1 = 'I like turtles'
    string2 = 'I like turkeys'
    assert hamming_distance(string1, string2) == 3

def test_3():
    string1 = 'a man a plan a canal'
    string2 = 'a man a plan sobanal'
    assert hamming_distance(string1, string2) == 3

def test_4():
    string1 = 'hamming and cheese'
    string2 = 'Hamming and Cheese'
    assert hamming_distance(string1, string2) == 2

def test_5():
    string1 = 'espresso'
    string2 = 'Expresso'
    assert hamming_distance(string1, string2) == 2

def test_6():
    string1 = 'old father, old artificer'
    string2 = 'of my soul the uncreated'
    assert hamming_distance(string1, string2) == 24



#https://www.codewars.com/kata/5410c0e6a0e736cf5b000e69/train/python